contrib/pyln-proto/pyln/proto/message/array_types.py

```diff
@@ -41,6 +41,13 @@ def val_to_str(self, v: List[Any], otherfields: Dict[str, Any]) -> str:
         s = ','.join(self.elemtype.val_to_str(i, otherfields) for i in v)
         return '[' + s + ']'
 
+    def val_to_py(self, v: Any, otherfields: Dict[str, Any]) -> Union[str, List[Any]]:
```

Comment:
Why is the `to_py()` method not part of the instances themselves, but rather the type class? Seems like a weird indirection.

Reply:
It'd also mean that the instance type (type of `v`) can extend the native python type and just work as one, without having to have an explicit conversion.